File: Com.Fubei.Api.OpenApi/Biz/AgentApi/Model/AEntities.cs
using System;
using System.Collections.Generic;
using System.Text;
using Com.Illuminati.Galileo.Model;
using Newtonsoft.Json;

namespace Com.Fubei.Api.OpenApi.Biz.AgentApi.Model
{
    // 以下为扩展参数
    #region 商品详情
    public class GoodsDetailEntity : BaseEntity
    {
        [JsonProperty("cost_price")]
        public int CostPrice { get; set; }

        [JsonProperty("receipt_id	")]
        public string ReceiptId { get; set; }

        [JsonProperty("goods_detail")]
        public List<GoodsDetailItemEntity> GoodsDetail { get; set; }
    }

    public class GoodsDetailItemEntity : BaseEntity
    {
        [JsonProperty("goods_id")]
        public string GoodsId { get; set; }

        [JsonProperty("goods_name")]
        public string GoodsName { get; set; }

        [JsonProperty("quantity")]
        public int Quantity { get; set; }

        [JsonProperty("price")]
        public int Price { get; set; }
    }
    #endregion

    #region 支付宝-花呗分期业务扩展

    public class AlipayExtendParamsEntity : BaseEntity
    {
        [JsonProperty("hb_fq_instalment")]
        public int? HbFqInstalment { get; set; }

        [JsonProperty("hb_fq_num")]
        public int? HbFqNum { get; set; }

        [JsonProperty("hb_fq_seller_percent")]
        public int? HbFqSellerPercent { get; set; }
    }

    #endregion

    #region 活动优惠列表
    public class APaymentItemEntity : BaseEntity
    {
        [JsonProperty("type")]
        public string Type { get; set; }

        [JsonProperty("amount")]
        public decimal Amount { get; set; }
    }
    #endregion
}

File: Com.Fubei.OpenApi.Vb/Biz/AgentApi/Model/AEntities.vb
Imports Com.Illuminati.Galileo.Model
Imports Newtonsoft.Json

' 公共类
Namespace Biz.V2.Model
    
    Public Class GoodsDetailEntity
        Inherits BaseEntity

        <JsonProperty("cost_price")>
        Public Property CostPrice As Integer

        <JsonProperty("receipt_id" & vbTab)>
        Public Property ReceiptId As String

        <JsonProperty("goods_detail")>
        Public Property GoodsDetail As List(Of GoodsDetailItemEntity)
    End Class

    Public Class GoodsDetailItemEntity
        Inherits BaseEntity

        <JsonProperty("goods_id")>
        Public Property GoodsId As String

        <JsonProperty("goods_name")>
        Public Property GoodsName As String

        <JsonProperty("quantity")>
        Public Property Quantity As Integer

        <JsonProperty("price")>
        Public Property Price As Integer
    End Class

    Public Class AlipayExtendParamsEntity
        Inherits BaseEntity

        <JsonProperty("hb_fq_instalment")>
        Public Property HbFqInstalment As Integer?

        <JsonProperty("hb_fq_num")>
        Public Property HbFqNum As Integer?

        <JsonProperty("hb_fq_seller_percent")>
        Public Property HbFqSellerPercent As Integer?
    End Class

    Public Class APaymentItemEntity
        Inherits BaseEntity

        <JsonProperty("type")>
        Public Property Type As String

        <JsonProperty("amount")>
        Public Property Amount As Decimal
    End Class


    Public Class ASignPackageEntity
        Inherits BaseEntity

        <JsonProperty("appId")>
        Public Property AppId As String

        <JsonProperty("timeStamp")>
        Public Property TimeStamp As String

        <JsonProperty("nonceStr")>
        Public Property NonceStr As String

        <JsonProperty("package")>
        Public Property Package As String

        <JsonProperty("signType")>
        Public Property SignType As String

        <JsonProperty("paySign")>
        Public Property PaySign As String
    End Class
End Namespace
